includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -922,6 +923,25 @@ private function validate_cdata_for_node( DOMElement $element, $cdata_spec ) {
 				return [ 'code' => self::MANDATORY_CDATA_MISSING_OR_INCORRECT ];
 			}
 		}
+
+		// When the CDATA is expected to be JSON, ensure it's valid JSON.
+		if ( 'script' === $element->tagName && in_array( $element->getAttribute( 'type' ), [ 'application/json', 'application/ld+json' ], true ) ) {
+			if ( empty( $element->textContent ) ) {
+				return [
+					'code'    => self::INVALID_JSON_CDATA,
+					'message' => esc_html__( 'Expected JSON, got an empty value', 'amp' ),
```

Comment:
```suggestion
					'message' => __( 'Expected JSON, got an empty value', 'amp' ),
```

Escaping should be done at printing not storage. The validation error may be sent back via JSON where HTML entities should not be present.

Reply:
See note below about how having a more specific error code for this case could be preferable, as the error message could be translated for the user at display time.